ouroboros-consensus/src/Ouroboros/Consensus/Mempool/Impl.hs

```diff
@@ -21,21 +23,38 @@ import           Ouroboros.Storage.ChainDB.API
 import           Ouroboros.Consensus.Ledger.Abstract
 import           Ouroboros.Consensus.Ledger.Extended
 import           Ouroboros.Consensus.Mempool.API
+import           Ouroboros.Consensus.Mempool.TxSeq (TicketNo, TxSeq (..),
+                     appendTx, fromTxSeq, lookupByTicketNo, splitAfterTicketNo,
+                     zeroTicketNo)
+import qualified Ouroboros.Consensus.Mempool.TxSeq as TxSeq
 import           Ouroboros.Consensus.Util (repeatedly)
+import           Ouroboros.Consensus.Util.STM (onEachChange)
+import           Ouroboros.Consensus.Util.ThreadRegistry (ThreadRegistry)
 
 {-------------------------------------------------------------------------------
   Top-level API
 -------------------------------------------------------------------------------}
 
-openMempool :: (MonadSTM m, ApplyTx blk)
-            => ChainDB m blk hdr
+openMempool :: ( MonadAsync m
+               , MonadFork m
+               , MonadMask m
+               , MonadSTM m
+               , ApplyTx blk
+               )
+            => ThreadRegistry m
```

Comment:
Ha, I see you already discovered we use `ThreadRegistry` for background threads :+1: